From: Nick Rosbrook <nick.rosbrook@canonical.com>
Date: Thu, 26 May 2022 14:32:20 -0400
Subject: sd-hwdb: include sys/stat.h in hwdb-internal.h

Origin: upstream, https://github.com/systemd/systemd/commit/9745b51c73
Bug-Ubuntu: https://launchpad.net/bugs/1988078
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Include this header to fix errors when including hwdb-internal.h:
  ../src/libsystemd/sd-hwdb/hwdb-internal.h:16:21: error: field ‘st’ has incomplete type
     16 |         struct stat st;
---
 src/libsystemd/sd-hwdb/hwdb-internal.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libsystemd/sd-hwdb/hwdb-internal.h b/src/libsystemd/sd-hwdb/hwdb-internal.h
index 5ddc221..62d27f7 100644
--- a/src/libsystemd/sd-hwdb/hwdb-internal.h
+++ b/src/libsystemd/sd-hwdb/hwdb-internal.h
@@ -2,6 +2,7 @@
 #pragma once
 
 #include <stdint.h>
+#include <sys/stat.h>
 
 #include "def.h"
 #include "hashmap.h"
